File: java/L100/LeetCode0169.java
package L100;

import java.util.HashMap;
import java.util.Map;

public class LeetCode0169 {
    public int majorityElement(int[] nums) {
        int N = -1;
        Map<Integer,Integer> mp = new HashMap<>();
        for (int x : nums) {
            int c = mp.getOrDefault(x, 0) + 1;
            if (c > nums.length / 2) {
                N = x;
                break;
            }
            mp.put(x, c);
        }
        return N;
    }
}

File: cpp/L100/LeetCode0169.cc
#include <vector>
using std::vector;

class Solution {
public:
    /**
     * 摩尔投票法找绝对众数（count > n / 2)
     */
    int majorityElement(vector<int>& nums) {
        int x{-1}, count{0};
        for (int num : nums) {
            if (count == 0) {
                x = num;
                count = 1;
            } else if (x == num) {
                count += 1;
            } else {
                count -= 1;
            }
        }
        return x;
    }
};